### PR TITLE
added exponential backoff and made depth_limit_at_nsq as command line_argument

### PR DESCRIPTION
**Pull request by praneeth36**
Connected #10

### CODE REVIEW

File: logagg/collect/collector.py [hunk 1]
```diff
@@ -155,7 +164,17 @@ def send_to_nsq(self):
 
             try:
                 if should_push:
+                    counter = 0
+                    self.SLEEP_TIME = 1
                     while 1:
+                        if counter < 5 and counter > 0:
+                            self.SLEEP_TIME += 1
```
Comment (bharathnfer) [inline logagg/collect/collector.py]: @praneeth36 exponential back off would mean the sleep time would be multiplied by 2, not added by one.

File: logagg/collect/collector.py [hunk 2]
```diff
@@ -116,9 +118,16 @@ def check_depth_at_nsq(self):
                 for record in topics:
                     topic_name = record['topic_name']
                     if self.nsqtopic == topic_name:
-                        depth_val = record.get('depth')
+                        channels = record.get('channels', [])
+                        if channels:
+                            for channel in channels:
+                                channel_name = channel.get('channel_name')
+                                if self.nsqchannel == channel_name:
+                                    depth_val = channel.get('depth')
```
Comment (bharathnfer) [inline logagg/collect/collector.py]: @praneeth36 so each channel has it's own depth now?

Comment (praneeth36) [reply logagg/collect/collector.py]: @bharathnfer 
- Yes, each channel has it's own depth. We are checking the depth at channel message queue.
- If the channel is not available, will limit the depth value at the topic message queue.

Comment (bharathnfer) [reply logagg/collect/collector.py]: @praneeth36 oh ok got it, thanks!

**Pull request merged**